dask_ml/model_selection/_incremental.py

```diff
@@ -599,20 +614,94 @@ def score(self, X, y=None):
         return self.scorer_(self.best_estimator_, X, y)
 
 
-class IncrementalSearch(BaseIncrementalSearch):
+class AdaptiveStopOnPlateauSearchCV(BaseIncrementalSearchCV):
```

Comment:
Inconsistent with the name in the `__init__.py`.

Reply:
-1 on this name.  I think that it is far too long and technical.  I think that it would scare non-techncial users.

Reply:
`AdaptiveStopOnPlateauSearchCV` is private (or at least that's what I intend). It is intended for use by other developers who have an adaptive algorithm they'd like to implement and also have stop on plateau (e.g., #221).

Here's the structure I intended:

* `BaseIncrementalSearchCV`: private.
    * Use: base class for all incremental searches
* `AdaptiveStopOnPlateauSearchCV`: private, inherits from `BaseIncrementalSearchCV`
    * Use: allow other devs easily create adaptive algorithms that stop on plateau (by overriding `_adapt`)
* `InverseTimeDecaySearchCV`: public, inherits from `AdaptiveStopOnPlateauSearchCV`

Reply:
I think that something like the implementation of `IncrementalSearch` in master should be public.  I think that `InverseTimeDecaySearchCV` is also probably an overly-technical name.

I recommend that we open an issue where we can discuss name changes to what is in master today and future restructurings.  In general I recommend doing this before investing much work.

Reply:
Indeed. The changes to `info` and `cv_results_`, etc. are useful in their own right. Those shouldn't be held up by a naming discussion.